### PR TITLE
Update Newsletter Template Entity

### PR DESCRIPTION
**Pull request by DmitryTsymbal**
**Description**
This PR with a test that updates Newsletter Template

**Steps:**
1 - Login as Admin
2 - Navigate to MARKETING > Newsletter Template
3 - Click the "Add New Template" button
4 - Fill in all data according to data set
5 - Save the template
6 - Making sure, that template was created
7 - Update Created Newsletter Template
8 - Perform Assertions

**Fixed Issues**
#470 Convert UpdateNewsletterTemplateTest to MFTF

### CODE REVIEW

File: app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        <actionGroup ref="AdminSearchNewsletterTemplateOnGridActionGroup" stepKey="findCreatedNewsletterTemplate">
+            <argument name="name" value="Updated Newsletter Template"/>
+            <argument name="subject" value="Updated Newsletter Subject"/>
+        </actionGroup>
+        <!-- Assert Created Newsletter on Grid -->
+        <actionGroup ref="AssertAdminCreatedNewsletterTemplateInGridActionGroup" stepKey="assertNewsletterInGrid">
+            <argument name="name" value="{{updatedNewsletter.name}}"/>
+            <argument name="subject" value="{{updatedNewsletter.subject}}"/>
+        </actionGroup>
+        <!-- END TEST BODY -->
```
Comment (rogyar) [inline app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: There's one more step in the original MTF tests that checks the updated template data

https://github.com/magento/magento2/blob/d1fb65c0f76947a78e28206a057d6597e4e32004/dev/tests/functional/tests/app/Magento/Newsletter/Test/Constraint/AssertNewsletterForm.php#L28

So, there were three constraints:
```xml
            <constraint name="Magento\Newsletter\Test\Constraint\AssertNewsletterSuccessCreateMessage" />
            <constraint name="Magento\Newsletter\Test\Constraint\AssertNewsletterInGrid" />
            <constraint name="Magento\Newsletter\Test\Constraint\AssertNewsletterForm" />
```
In the current test we have only two constraints.

Comment (DmitryTsymbal) [reply app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: Hello @rogyar , I have added Newsletter Form Assertion into my test. Could you please review it?
Thanks!

File: app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AssertAdminCreatedNewsletterTemplateMessageActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertAdminCreatedNewsletterTemplateMessageActionGroup">
```
[the remark this answers is not in the document]
Comment (DmitryTsymbal) [reply app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AssertAdminCreatedNewsletterTemplateMessageActionGroup.xml]: @rogyar , nice notice, thanks!
Re-used ☺️

File: app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminMarketingOpenNewsletterTemplateFromGridActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+
```
Comment (lbajsarowicz) [inline app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminMarketingOpenNewsletterTemplateFromGridActionGroup.xml]: Redundant space.

File: app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminNavigateToCreateNewsletterTemplatePageActionGroup.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+
```
Comment (lbajsarowicz) [inline app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminNavigateToCreateNewsletterTemplatePageActionGroup.xml]: Redundant space.

File: app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            <features value="Newsletter"/>
+            <stories value="Newsletter Updating"/>
+            <title value="Newsletter Updating Test"/>
+            <description value="Admin should be able update created Newsletter Template"/>
+            <group value="reports"/>
+            <group value="mtf_migrated"/>
+        </annotations>
+
+        <before>
+            <actionGroup ref="LoginAsAdmin" stepKey="LoginAsAdmin"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: `AdminLoginActionGroup`

File: app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            <!-- Navigate To MARKETING > Newsletter Template -->
```
Comment (lbajsarowicz) [inline app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: Remove comments as these appear on Test Results as real steps even if you remove surrounding real steps.

File: app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+                <argument name="menuUiId" value="{{AdminMenuMarketing.dataUiId}}"/>
+                <argument name="submenuUiId" value="{{AdminMenuMarketingCommunicationsNewsletterTemplate.dataUiId}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminSearchNewsletterTemplateOnGridActionGroup" stepKey="findCreatedNewsletterTemplateInGrid">
+                <argument name="name" value="{{updatedNewsletter.name}}"/>
+                <argument name="subject" value="{{updatedNewsletter.subject}}"/>
+            </actionGroup>
+            <actionGroup ref="AdminMarketingOpenNewsletterTemplateFromGridActionGroup" stepKey="openTemplate"/>
+            <actionGroup ref="AdminMarketingDeleteNewsletterTemplateActionGroup" stepKey="deleteTemplate"/>
+            <actionGroup ref="logout" stepKey="logout"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: `AdminLogoutActionGroup`

File: app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminMarketingOpenNewsletterTemplateFromGridActionGroup.xml (hunk 3 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminMarketingOpenNewsletterTemplateFromGridActionGroup.xml]: Done

File: app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminNavigateToCreateNewsletterTemplatePageActionGroup.xml (hunk 4 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Newsletter/Test/Mftf/ActionGroup/AdminNavigateToCreateNewsletterTemplatePageActionGroup.xml]: Done

File: app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml (hunk 5 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: Done

File: app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml (hunk 6 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: Done

File: app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml (hunk 7 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Newsletter/Test/Mftf/Test/AdminMarketingNewsletterTemplateUpdateTest.xml]: Done

**Pull request merged**